sopel/modules/safety.py

```diff
@@ -26,56 +27,107 @@
+    | domain\\_blocklist\\_url | https://example.com/bad-hosts.txt | Optional hosts-file formatted domain blocklist to use instead of StevenBlack's. |
     """
-    config.define_section('safety', SafetySection)
-    config.safety.configure_setting(
-        'enabled_by_default',
-        "Enable URL safety in channels that don't specifically disable it?",
+    settings.define_section("safety", SafetySection)
+    settings.safety.configure_setting(
+        "default_mode",
+        (
+            "Which mode should be used in channels that haven't specifically set one?"
+            "\n({})".format("/".join(SAFETY_MODES))
+        ),
+        default="on",
     )
-    config.safety.configure_setting(
+    settings.safety.configure_setting(
         'known_good',
-        'Enter any domains to whitelist',
+        "Enter any domains to allowlist",
     )
-    config.safety.configure_setting(
+    settings.safety.configure_setting(
         'vt_api_key',
         "Optionally, enter a VirusTotal API key to improve malicious URL "
-        "protection.\nOtherwise, only the StevenBlack list will be used."
+        "protection.\nOtherwise, only the configured hosts list will be used.",
+    )
+    settings.safety.configure_setting(
+        'domain_blocklist_url',
+        "Optionally, provide the URL for a hosts-file formatted domain "
+        "blocklist to use instead of StevenBlack's.",
     )
 
 
 def setup(bot):
-    bot.config.define_section('safety', SafetySection)
+    bot.settings.define_section("safety", SafetySection)
+
+    if bot.settings.safety.default_mode is None:
+        bot.settings.safety.default_mode = "on"
+    # migrate from enabled_by_default to default_mode
+    if bot.settings.safety.enabled_by_default is not None:
+        if bot.settings.safety.enabled_by_default.lower() not in [
+            "1",
+            "enable",
+            "enabled",
+            "on",
+            "true",
+            "y",
+            "yes",
+        ]:
+            bot.settings.safety.default_mode = "off"
+        bot.settings.safety.enabled_by_default = None
+        LOGGER.info(
+            "config: enabled_by_default is deprecated, please use default_mode=%s",
+            bot.settings.safety.default_mode,
+        )
+
+    if bot.settings.safety.vt_api_key is not None:
+        requests_session.headers = {
+            "x-apikey": bot.settings.safety.vt_api_key,
+        }
 
     if 'safety_cache' not in bot.memory:
         bot.memory['safety_cache'] = tools.SopelMemory()
     if 'safety_cache_lock' not in bot.memory:
         bot.memory['safety_cache_lock'] = threading.Lock()
-    for item in bot.config.safety.known_good:
+    for item in bot.settings.safety.known_good:
         known_good.append(re.compile(item, re.I))
 
-    old_file = os.path.join(bot.config.homedir, 'malwaredomains.txt')
+    update_local_cache(bot, init=True)
+
+
+def update_local_cache(bot, init: bool = False):
```

Comment:
Since you are adding type hint, I invite you to also type `bot`: it can be either a `sopel.bot.Sopel` or a `sopel.bot.SopelWrapper`.